src/plugins/irc-events/link.js

```diff
@@ -146,7 +147,23 @@ function handlePreview(client, msg, preview, res) {
 		return emitPreview(client, msg, preview);
 	}
 
-	storage.store(res.data, res.type.replace("image/", ""), (uri) => {
+	// Get the correct file extension for the provided content-type
+	// This is done to prevent user-input being stored in the file name (extension)
+	const extension = mime.extension(res.type);
+
+	if (!extension) {
+		// For link previews, drop the thumbnail
+		// For other types, do not display preview at all
+		if (preview.type === "link") {
+			preview.thumb = "";
+		} else {
+			return;
+		}
+
+		return emitPreview(client, msg, preview);
+	}
+
+	storage.store(res.data, extension, (uri) => {
```

Comment:
Any chance you could add a test or 2 in https://github.com/thelounge/lounge/blob/master/test/plugins/link.js ?